pymc3/distributions/distribution.py

```diff
@@ -304,14 +304,14 @@ def draw_values(params, point=None, size=None):
     missing_inputs = set(params)
     while to_eval or missing_inputs:
         if to_eval == missing_inputs:
-            raise ValueError('Cannot resolve inputs for {}'.format([str(j) for j in to_eval]))
+            raise ValueError('Cannot resolve inputs for {}'.format([str(params[j]) for j in to_eval]))
         to_eval = set(missing_inputs)
         missing_inputs = set()
         for param in to_eval:
             try:  # might evaluate in a bad order,
                 evaluated[param] = _draw_value(params[param], point=point, givens=givens.values(), size=size)
-                if any(param in j for j in named_nodes_children.values()):
-                    givens[param.name] = (params[param], evaluated[param])
+                if any(params[param] in j for j in named_nodes_children.values()):
```

Comment:
It seems you committed this while I was commenting and my comment got immediately outdated. However I still think this line is confusing, You're trying to see if the node `params[param]` is a child of some other named node. If `params[param]` were to be a named node, that information should be available in the dictionary `named_nodes_parents`. If `params[param]` were not to be a named node, then it would not be registered neither in the `named_nodes_parents` nor the `named_nodes_children` dictionaries.
If `params[param]` is a named node, you should be able to replace this line by:
`if named_nodes_parents[params[param]]:`
If `params[param]` is not a named node, then I think it shouldn't bee added to `givens` but I may be overlooking something.

Reply:
This is much nicer, thank you!

Reply:
(also just checked out travis, and your suggestion will also fix failing tests)